src/tests/Microsoft.PowerFx.Interpreter.Tests/UserInfoTests.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Tests;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Types;
+using Xunit;
+
+namespace Microsoft.PowerFx.Interpreter.Tests
+{
+    public class UserInfoTests : PowerFxTest
+    {
+        [Theory]
+        [InlineData("FirstName LastName", "a@b.com")]
+        [InlineData("c203b79b-b985-42f0-b523-c10eb64387c6", null)]
+        [InlineData(null, null)]
+        [InlineData("", "")]
+        public async Task UserInfoObjectTest(string fullName, string email)
+        {            
+            var userInfo = new UserInfo
+            {
+                FullName = fullName,
+                Email = email
+            };
+
+            var userInfoType = RecordType.Empty()
+                .Add("FullName", FormulaType.String)
+                .Add("Email", FormulaType.String);
+
+            SymbolTable symbol = new SymbolTable();
+            symbol.AddHostObject("UserInfo", userInfoType, GetUserInfoObject);
```

Comment:
>symbol.AddHostObject("UserInfo", userInfoType, GetUserInfoObject); [](http://example.com/codeflow?start=12&length=66)

We should wrap this as a convenience method in Core. 

Extension like:
symbol.AddUserObject();

or 
config.SymbolTable.AddUserObject();

which adds to the symbol table, and then at runtime, they call the RuntimeConfig extension on line 41

So GetUserInfoObject gets pulled into Core as a default implementation, but note that 
- uses IUserInfo
- does not need to use TypeMarshallerCache (since that's only in interpreter). We can just manually create the FormulaVAlue

Reply:
update this. thanks